torch/fx/experimental/symbolic_shapes.py

```diff
@@ -269,20 +326,68 @@ def eval(cls, base, divisor):
     'sym_sqrt': lambda a: sympy.sqrt(a),
 }
 
+sizes_strides_methods = {
+    'is_non_overlapping_and_dense': lambda *args: IsNonOverlappingAndDenseIndicator(*args),
+}
+
+def eval_is_non_overlapping_and_dense(sizes, strides):
+    dim = len(sizes)
+
+    # Short-circuits for tensors of rank one, which are
+    # non-overlapping and "dense" if their stride is one
+    if dim == 1:
+        return strides[0] == 1
+
+    # Checks that there exists a permutation of the strides s.t. the tensor would be contiguous
+    # Sorts (length, stride) pairs by stride
+    lengths_and_strides = sorted(
+        tuple(zip(sizes, strides)), key=operator.itemgetter(1)
+    )
+
+    expected_stride = 1
+    for length, stride in lengths_and_strides:
+
+        if length == 1:
+            continue
```

Comment:
Why is the cpp code doing `return True` here?

Reply:
Ho the cpp code special cases the sorting to ensure that dimensions of size 0 or 1 are at the end. So they know that by this point you can stop.
In this case we don't do this special casing in sorting and so we have to continue.

This might deserve a comment :)

Reply:
So... I was naughty and this is actually copy pasted from the non overlapping and dense implementation in PrimTorch. I should probably dedupe them (this one being canonical); the PrimTorch implementation is annoying because it requires full tensors. I am not actually using it in this PR, and can put it off later. What would you prefer?

Reply:
That can be done later, it's fine. Can you leave a TODO then?